# Velocity2020
NERC Velocity Project analyses using Sanger genomes

Here I'll curate the variant calling pipeline and analyses undertaken using the Lepidoptera genomes generated by Sanger in 2019/2020. 


## Genome

Aphantopus hyperantus (Ringlet) was the first genome available ([NCBI link](https://www.ncbi.nlm.nih.gov/assembly/GCA_902806685.1)), so the pipeline will be set up with this species. 

## WGS

Whole genome resequencing data was generated for 38 & 40 modern individuals (sampled 2016-2017 & 2019) from a core and expanding population. Museum data was generated from 48 individuals + resequencing of a subset of individuals to increase read coverage. 

## Pipeline

### 1. Demultiplex and Adapter trimming

Modern samples arrive demultiplexed by the sequencing facility, but Museum samples need to be demultiplexed. 

### 2. Map to Reference Genome

### 3. MapDamage: correct for Cytosine deamination in museum data

#### 3a. MapDamage run on museum data

#### 3b. Downsample modern data to the same coverage as in the museum samples


### 4. ANGSD

#### 4a. Set up filters (including min and maxDepth)

#### 4b. Call GL

#### 4c. SFS

#### 4d. Compare downsampled data to full dataset for modern pops (SFS and GL)

#### 4e. Population structure (PCA)

#### 4f. Diversity stats 

#### 4g. Outliers

#### 4h. LD analyses


### 5. Map outlier loci

Map outlier loci, identify candidates in the area

Synteny between modern and museum samples. 





